src/base/static/components/form-field/index.js

```diff
@@ -1,354 +1,114 @@
 import React, { Component } from "react";
 import PropTypes from "prop-types";
-import { fromJS, List as ImmutableList, Map as ImmutableMap } from "immutable";
+import { Map as ImmutableMap } from "immutable";
 import classNames from "classnames";
 
-import {
-  TextField,
-  TextareaField,
-  DropdownField,
-  DatetimeField,
-  GeocodingField,
-  AddAttachmentButton,
-  BigRadioField,
-  BigCheckboxField,
-  InputFormSubmitButton,
-  RichTextareaField,
-  MapDrawingToolbar,
-  AutocompleteComboboxField,
-  CustomUrlToolbar,
-  BigToggleField,
-  PublishControlToolbar,
-  RangeSliderWithLabel,
-} from "../form-field-types";
-
-import "./form-field.scss";
+import fieldDefinitions from "./field-definitions";
 import { inputForm as messages } from "../messages.js";
-import {
-  mayHaveAnyValue,
-  mustHaveSomeValue,
-  mustHaveUniqueUrl,
-} from "./validators";
 import constants from "../constants";
 
+import "./form-field.scss";
+
 const Util = require("../../js/utils.js");
 
 class FormField extends Component {
-  shouldComponentUpdate(nextProps) {
-    return (
-      nextProps.isInitializing ||
-      nextProps.showValidityStatus ||
-      nextProps.updatingField === this.props.config.name
-    );
-  }
-
   componentWillMount() {
-    this.validator = this.getValidator(this.props.config);
```

Comment:
It doesn't seem like there is anything in this `componentWillMount` that is a subscription or async fetch. Why not move it into the constructor?